src/AudioAnalysisTools/Tracks/OnebinTrackAlgorithm.cs

```diff
@@ -66,8 +71,22 @@ public static class OnebinTrackAlgorithm
             int binCount = sonogramData.GetLength(1);
             int nyquist = sonogram.NyquistFrequency;
             double binWidth = nyquist / (double)binCount;
-            int minBin = (int)Math.Round(parameters.MinHertz.Value / binWidth);
-            int maxBin = (int)Math.Round(parameters.MaxHertz.Value / binWidth);
+
+            // set lower frequency bins of the search band
+            int minSearchBin = (int)Math.Floor(parameters.SearchbandMinHertz.Value / binWidth);
+            if (minSearchBin < 1)
+            {
+                minSearchBin = 1;
+            }
+
+            // set top search bin allowing for the top sideband.
+            int maxSearchBin = (int)Math.Floor(parameters.SearchbandMaxHertz.Value / binWidth) - 1;
```

Comment:
The incorrect parameters `SearchbandMinHertz` and `SearchbandMaxHertz` are still being used here

Reply:
I have made the corrections as requested. I have also made an extensive class summary for the OneBinTrackAlgorithm class i.e. the class used to find whistles. I hope this will be helpful in understanding the algorithm and explaining the "magic numbers". In fact all the track algorithms  require that a lot of thought be given to the choice of sample rate, frame size and frame step. The typical values are not always appropriate depending on the target call.